### DISCUSSION TITLE
Using CefSharp with maui

### DISCUSSION THREAD

**Discussion by CodeDoctorDE** [cefsharp/CefSharp#4037] [General]
Hello,

is there a way to integrate cefsharp with a maui application?

**Comment by amaitland** [comment]
From the little I know about MAUI there's only a single hwnd per window, so a MAUI specific implementation would be required (isn't one currently).

As CefSharp only runs on Windows and MAUI is crossplat it's not clear there would be much call for a MAUI specific implementation.

Are you creating a Windows only MAUI app?

**Reply by CodeDoctorDE** [reply]
No I try to make it crossplatform

**Reply by amaitland** [reply]
Maybe checkout MAUI blazor

**Reply by CodeDoctorDE** [reply]
can i use the cef in maui blazor?

**Comment by fabriceleal** [comment]
Are there any plans to bring CefSharp to MAUI even if it's for the Windows platform alone?

**Reply by amaitland** [reply]
MAUI already has a [WebView](https://learn.microsoft.com/en-us/dotnet/api/microsoft.maui.controls.webview?view=net-maui-7.0), what would be the advantage of using `CefSharp` exactly?

**Reply by fabriceleal** [reply]
The API that MAUI exposes for the native WebView lacks some features that I require, the most glaring of them the lack of a [RegisterJsObject](http://cefsharp.github.io/api/71.0.0/html/M_CefSharp_Wpf_ChromiumWebBrowser_RegisterJsObject.htm) like method.

**Reply by amaitland** [reply]
Personally I have no need for a `MAUI` version at this time. I'd have to take on a paying job that required a `MAUI` implementation to justify the time required.

Someone from the community is welcome to take up the challenge.

**Reply by amaitland** [reply]
> The API that MAUI exposes for the native WebView lacks some features that I require, the most glaring of them the lack of a [RegisterJsObject](http://cefsharp.github.io/api/71.0.0/html/M_CefSharp_Wpf_ChromiumWebBrowser_RegisterJsObject.htm) like method.

See https://github.com/dotnet/maui/discussions/12009#discussioncomment-5210723 for my proposal to implement this in `MAUI` using the current `WebView`.
